app/schemas/federation_entities.json

```diff
@@ -0,0 +1,366 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "https://diasporafoundation.org/schemas/federation_entities-1.0.0.json",
```

Comment:
I would prefer to keep the schema in a separate project. In theory the schema can be used in a different compatible implementation which is written in a different language and that has nothing to do with the ruby implementation of diaspora federation. Currently it is hardly possible that someone implements compatible software without reading our ruby source code, but in future it should be possible to implement protocol by using specifications and schemas and without touching ruby implementation.

But then the question raises: how should we deliver the schema to the `diaspora_federation` gem? We could create a gem for this, use git submodule in this repository or even download schema by its id URL. Using a gem is simple and robust, but using id URL to download schemas is better since it uses the "built-in" schema distribution approach (the fact that it is accessible via that URL) and doesn't need any extra distribution efforts. However if we choose the latter we'll need some caching method so that we don't have to download the schema on each rspec execution.